src/UserManager.php

```diff
@@ -209,33 +218,36 @@ static function (array $user) use ($userIdClaim) {
       $users = $this->userStorage->loadByProperties([$userIdField => $userIdsToKeep]);
       foreach ($users as $user) {
         $this->logger->info($this->t('Retaining user @name.', ['@name' => $user->label()]));
-        unset($cachedUserIdsForDeletion[$user->id()]);
+        unset($this->userIds[$user->id()]);
       }
-      $this->cacheUserIdsForDeletion($cachedUserIdsForDeletion);
     }
   }
 
   /**
    * {@inheritdoc}
    */
   public function commitDeletionList(): void {
-    $method = $this->moduleConfig->get('user_cancel_method');
+    // user_cancel_block: Account will be blocked and will no longer be able
+    // to log in. All of the content will remain attributed to the username.
+    $method = $this->moduleConfig->get('drupal')['user_cancel_method'] ?? 'user_cancel_block';
     $deletedUserIds = [];
-    $userIds = $this->getCachedUserIdsForDeletion();
-    foreach ($userIds as $userId) {
+
+    foreach ($this->userIds as $userId) {
       user_cancel([], $userId, $method);
       $deletedUserIds[] = $userId;
     }
 
-    $this->logger->info($this->formatPlural(
-      count($deletedUserIds),
-      'One user to be deleted',
-      '@count users to be deleted'
-    ));
-    if ($this->options['debug'] ?? FALSE) {
-      $users = $this->userStorage->loadMultiple($deletedUserIds);
-      foreach ($users as $user) {
-        $this->logger->debug(sprintf('User to be deleted: %s (#%s)', $user->label(), $user->id()));
+    if (0 !== count($deletedUserIds)) {
+      $this->logger->info($this->formatPlural(
+        count($deletedUserIds),
+        'One user to be deleted',
+        '@count users to be deleted'
+      ));
+      if ($this->options['debug'] ?? FALSE) {
+        $users = $this->userStorage->loadMultiple($deletedUserIds);
+        foreach ($users as $user) {
+          $this->logger->debug(sprintf('User to be deleted: %s (#%s)', $user->label(), $user->id()));
+        }
```

Comment:
The empty list is also a list:

```suggestion
    $this->logger->info($this->formatPlural(
      count($deletedUserIds),
      'One user to be deleted',
      '@count users to be deleted'
    ));
    if ($this->options['debug'] ?? FALSE) {
      $users = $this->userStorage->loadMultiple($deletedUserIds);
      foreach ($users as $user) {
        $this->logger->debug(sprintf('User to be deleted: %s (#%s)', $user->label(), $user->id()));
      }
```